uxarray/grid.py

```diff
@@ -0,0 +1,279 @@
+"""uxarray grid module."""
+import os
+import xarray as xr
+import numpy as np
+from pathlib import PurePath
+
+
+class Grid:
+    """A class for uxarray grid object.
+
+    Examples
+    ========
+    import uxarray as ux
+    # open an exodus file with Grid object
+    mesh = ux.Grid("filename.g")
+
+    # save as ugrid file
+    mesh.saveas("outfile.ug")
+    """
+
+    # Import read/write methods from files in this folder
+    from ._exodus import read_exodus, write_exodus
+    from ._ugrid import read_ugrid, write_ugrid
+    from ._shpfile import read_shpfile
+    from ._scrip import read_scrip
+
+    def __init__(self, *args, **kwargs):
+        """Initialize grid variables, decide if loading happens via file, verts
+        or gridspec If loading from file, initialization happens via the
+        specified file.
+        Args: input file name with extension as a string
+              vertex coordinates that form one face.
+        kwargs: can be a dict initializing specific variables: kwargs = {"concave" : True, "islatlon" : True"}
+
+        Raises:
+            RuntimeError: File not found
+        """
+        # initialize possible variables
+        for key, value in kwargs.items():
+            if key == "latlon":
+                self.islatlon = value
+            elif key == "gridspec":
+                self.gridspec = value
+
+        self.filepath = None
```

Comment:
To increase readability for the instance attributes for Grid class, I'd suggest moving these attributions to the very beginning of the `__init__` functions.